### PR TITLE
feat(ST-87): add endpoint for enrollment of student to classroom

### PR DESCRIPTION
**Pull request by ghost**
## Description of Change
- Created an enrollment module and add that on app.module.ts file
- Created enrollment.dtos.ts, enrollment.controller.ts and enrollment.service.ts file.
- Added logic in the service layer 
- Add dto for creating and for response of enrollment

## Associated Ticket Link(s)

- https://trello.com/c/tn7Kz2CM

## Screenshots / Screen Recordings
N/A

### CODE REVIEW

File: src/enrollments/enrollments.dtos.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  studentId!: number;
+
+  @IsNumber()
+  classroomId!: number;
+
+  constructor(id: number, studentId: number, classroomId: number) {
+    this.id = id;
+    this.studentId = studentId;
+    this.classroomId = classroomId;
+  }
```
Comment (ishmam-iqbal-sazim) [inline src/enrollments/enrollments.dtos.ts]: Do you need all this to be sent via the response upon successful enrollment?

File: src/enrollments/enrollments.service.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    const existingEnrollment = await this.enrollmentRepository.findOne({ student, classroom });
+    if (existingEnrollment) {
+      throw new ConflictException(`Student is already enrolled in this classroom`);
+    }
+
+    const enrollment = this.enrollmentRepository.create({ student, classroom });
+
+    await this.em.persistAndFlush(enrollment);
+
+    return new EnrollmentResponseDto(enrollment.id, student.id, classroom.id);
```
Comment (ishmam-iqbal-sazim) [inline src/enrollments/enrollments.service.ts]: Is this doing what a serializer should be doing?

File: src/enrollments/enrollments.controller.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+@Controller("enrollments")
+@UseGuards(JwtAuthGuard, RolesGuard)
+export class EnrollmentsController {
+  constructor(private readonly enrollmentsService: EnrollmentsService) {}
+
+  @Roles(EUserType.TEACHER)
+  @Post()
+  enrollStudent(@Body() createEnrollmentDto: CreateEnrollmentDto): Promise<EnrollmentResponseDto> {
+    return this.enrollmentsService.enrollStudent(createEnrollmentDto);
+  }
```
Comment (ishmam-iqbal-sazim) [inline src/enrollments/enrollments.controller.ts]: I think there should be no enrollments controller at all as it is the junction/pivot table.
Rather this should be an endpoint in the classrooms controller, for example: `classrooms/enroll`

Comment (ghost) [reply src/enrollments/enrollments.controller.ts]: Bhaiya I did this for separating enrollments logic from classsroom. There already loads going on in classroom controller, hence I did this for ease.

File: src/enrollments/enrollments.dtos.ts (hunk 1 above)
Comment (ghost) [reply src/enrollments/enrollments.dtos.ts]: I thought I need studentId and classroomId for sending notification via mail. But as I already implemented the mail logic I don't need this response, instead I will return true if the enrollment is successful

File: src/enrollments/enrollments.service.ts (hunk 2 above)
Comment (ghost) [reply src/enrollments/enrollments.service.ts]: Yes. It was shaping the response

File: src/enrollments/enrollments.service.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import { CreateEnrollmentDto } from "./enrollments.dtos";
+import { EnrollmentsRepository } from "./enrollments.repository";
+
+@Injectable()
+export class EnrollmentsService {
+  constructor(
+    private readonly em: EntityManager,
+    private readonly studentRepository: StudentsRepository,
+    private readonly classroomRepository: ClassroomsRepository,
+    private readonly enrollmentRepository: EnrollmentsRepository,
```
Comment (ishmam-iqbal-sazim) [inline src/enrollments/enrollments.service.ts:17]: [NIT] 


```suggestion
    private readonly studentsRepository: StudentsRepository,
    private readonly classroomsRepository: ClassroomsRepository,
    private readonly enrollmentsRepository: EnrollmentsRepository,
```

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left a comment

**Pull request merged**